toolchain/lex/string_literal.cpp

```diff
@@ -127,7 +129,12 @@ auto StringLiteral::Lex(llvm::StringRef source_text)
     // escape sequences starting with a predictable character and not containing
     // embedded and unescaped terminators or newlines.
     switch (source_text[cursor]) {
+      case '\t':
+        // Tabs have extra validation.
+        content_needs_validation = true;
+        break;
       case '\\':
+        content_needs_validation = true;
```

Comment:
May as well move this inside the `if` below; a plain `\\` that's not part of an escape doesn't need validation.

Reply:
Done. I was thinking about this wrong, thinking it would suppress a diagnostic when this is actually relevant to raw string handling.